packages/backend/src/nest/storage/storage.service.ts

```diff
@@ -729,18 +750,55 @@ export class StorageService extends EventEmitter {
     return result
   }
 
-  public getAllUsers(): UserData[] {
-    const csrs = this.getAllEventLogEntries(this.certificatesRequestsStore.store)
-    this.logger('csrs count:', csrs.length)
-    const allUsers: UserData[] = []
+  /**
+   * Retrieve all users (using certificates and CSRs to determine users)
+   */
+  // TODO: Add test case around have both certificates and CSRs
+  public async getAllUsers(): Promise<UserData[]> {
+    const csrs = await this.certificatesRequestsStore.getCsrs()
+    const certs = await this.certificatesStore.getCertificates()
+    const allUsersByKey: Record<string, UserData> = {}
+
+    this.logger(`Retrieving all users. CSRs count: ${csrs.length} Certificates count: ${certs.length}`)
+
+    for (const cert of certs) {
```

Comment:
I'm guessing that retrieving both certs and csrs is for the case when one of those DBs are not fully synced? In any other case you can't have certificate without csr.

Reply:
This mirrors how we do things on the frontend and is helpful in the case where certs replicate before csrs.